### PR TITLE
Python 3.6+, Django 3+, lints

### PR DESCRIPTION
**Pull request by akx**
This PR:

* runs pyupgrade, flynt, etc. to make the codebase Python 3.6+
* runs django-codemod to upgrade things for Django 3+
* fixes up some old dev deps and other such CI things
* adds flake8 + isort linting (and fixes problems)

<del>
* adds types using Monkeytype 🐒 
* adds type checking using Mypy
</del>

I realized the above should probably be in a separate PR.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 The MIT License (MIT)
 
-Copyright (c) 2018 Kehittyvien Conien Suomi ry, Aarni Koskela
+Copyright (c) 2018-2021 Kehittyvien Conien Suomi ry, Aarni Koskela
```
Comment (japsu) [inline LICENSE]: I guess no-one of us has actually assigned copyright to KcS, so we should probably just list individual names here.

File: baikal/urls.py [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
+inspection_view = InspectionView.as_view(require_same_user=False, template_name='inspect.html')
 urlpatterns = [
-    url(r'^admin/', admin.site.urls),
-    url(r'^relinquish/(?P<pk>\d+)/$', RelinquishView.as_view(require_same_user=False), name='relinquish'),
-    url(r'^reserve/(?P<pk>\d+)/$', ReservationView.as_view(template_name='generic_form.html'), name='reserve'),
-    url(r'^inspect/(?P<pk>\d+)/(?P<key>.+?)/$', InspectionView.as_view(require_same_user=False, template_name='inspect.html'), name='inspect'),
-    url(r'^print/(?P<pk>\d+)/$', PrintView.as_view(template_name='generic_form.html'), name='print'),
-    url(r'^$', IndexView.as_view()),
+    path('admin/', admin.site.urls),
+    path('relinquish/<int:pk>/', RelinquishView.as_view(require_same_user=False), name='relinquish'),
```
Comment (japsu) [inline baikal/urls.py:11]: TIL

File: paikkala/fields.py [hunk 3]
```diff
@@ -39,7 +37,7 @@ def populate_reservation_statuses(self, program):
     def label_from_instance(self, obj):
         if obj in self.reservation_statuses:
             info = self.reservation_statuses[obj]
-            return force_text(self.label_format).format(
+            return str(self.label_format).format(
```
Comment (japsu) [inline paikkala/fields.py]: `self.format_label`?

**Review by japsu — APPROVED** [review]

File: baikal/urls.py (hunk 2 above)
Comment (akx) [reply baikal/urls.py:11]: About `path()`?

Comment (japsu) [reply baikal/urls.py:11]: `<int:`

**Pull request merged**